### DISCUSSION TITLE
[Question] Difference between ask and ask_stream?

### DISCUSSION THREAD

**Discussion by scmanjarrez** [acheong08/EdgeGPT#57] [Q&A] [answered]
Hi, first of all, amazing tool, thank you!

I was reading the code and saw `ask` and `ask_stream` functions, what is the difference? `ask` when you want to use it only for 1 question, and `ask_stream` if you want to have a conversation?

**Comment by acheong08** [comment, marked as answer]
Stream is a generator and ask is a normal function. Stream allows you to get data faster and use them as you get it. You can try it out via the CLI. `--no-stream` means to use `ask` and the default is `ask_stream`

**Comment by scmanjarrez** [comment]
Thank you!
